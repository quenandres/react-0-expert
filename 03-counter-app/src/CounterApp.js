import React from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {
    return (<>
    <h1>CounterApp</h1>
    <h2> { value } </h2>
    </>)
}

CounterApp.prototypes = {
    value: PropTypes.number.isRequired,
}

CounterApp.defaultProps = {
    value: 0
}

export default CounterApp;