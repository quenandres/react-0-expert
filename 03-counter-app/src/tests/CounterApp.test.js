import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('Pruebas <CounterApp />', () => {
    test('debe mostrar <CounterApp/> correctamente (hacer match con un snapshot) y sus valores por defecto', () => {
        const wrapper = shallow(<CounterApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar el valor por defecto de 100', () => {
        const wrapper = shallow(<CounterApp value={ 100 } />);
        const numero100 = wrapper.find('h2').text().trim();
        expect( numero100 ).toBe( '100' );
    });
    
    /* test('usar wrapper.find. tomando el elemento html donde se muestra el valor del contador', () => {
        
    });    */ 
});