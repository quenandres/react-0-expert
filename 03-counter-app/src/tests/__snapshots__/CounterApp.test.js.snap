// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <CounterApp /> debe mostrar <CounterApp/> correctamente (hacer match con un snapshot) y sus valores por defecto 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    0
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    0
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
